velox/functions/prestosql/Sequence.cpp

```diff
@@ -131,20 +131,39 @@ class SequenceFunction : public exec::VectorFunction {
 
   static void writeToElements(
       T* elements,
+      bool isDate,
       vector_size_t sequenceCount,
       DecodedVector* startVector,
       DecodedVector* stopVector,
       DecodedVector* stepVector,
       vector_size_t row) {
     auto start = startVector->valueAt<T>(row);
     auto stop = stopVector->valueAt<T>(row);
-    const int64_t step = (stepVector == nullptr)
-        ? (toInt64(stop) >= toInt64(start) ? 1 : -1)
-        : toInt64(stepVector->valueAt<T>(row));
+    auto step = getStep(toInt64(start), toInt64(stop), stepVector, row, isDate);
     for (auto i = 0; i < sequenceCount; ++i) {
       elements[i] = add(start, step * i);
     }
   }
+
+  static int64_t getStep(
+      int64_t start,
+      int64_t stop,
+      DecodedVector* stepVector,
+      vector_size_t row,
+      bool isDate) {
```

Comment:
nit: use early exist to simplify code
https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code 

```
if (!stepVector) {
     return (stop >= start ? 1 : -1);
}
auto step = stepVector->valueAt<int64_t>(row);
if(!isDate){
       return step;
}
  
// Handle Date
VELOX_USER_CHECK(step % kMillisInDay == 0, 
 "sequence step must be a day interval if start and end values are dates");
return step / kMillisInDay;
```